tests/subsys/power/power_mgmt/src/dummy_driver.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/printk.h>
+#include <zephyr/types.h>
+#include "dummy_driver.h"
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
```

Comment:
Don't wrap an entire C file in a preprocessor macro like this.  When you need to do that, there is a cmake macro for conditioning source files on kconfig symbols.

But in this case, surely you shouldn't even be building this test on systems without DEVICE_POWER_MANAGEMENT selected, right?  Just put a filter in the test case YAML file.  What are you trying to work around here?

Reply:
agree using cmake to avoid wrapping C file in a preprocessor macro